docs/template-development.md

```diff
@@ -0,0 +1,166 @@
+
+### Accessing parameters inside the template
+
+Additionally, we can also have a configuration called `parameters` which is an object with all the parameters that can be passed when generating the template:
+
+```json
+{
+  "name": "myTemplate",
+  "generator": {
+    "renderer": "react",
+    "supportedProtocols": "mqtt",
+    "parameters": {
+        "version": {
+          "description": "Override the version of your application provided under `info.version` location in the AsyncAPI document.",
+          "required": false
+        }
+    }
+  },
+  "dependencies": {
+    "@asyncapi/generator-react-sdk": "^0.2.25"
+  }
+}
+```
+
+The default version of your application is always fetched from your AsyncAPI document. The above configuration will help the template user to override the existing version with a new version on the command line.
+
+The changes done in the template will be as follows:
+
+Original:
+
+```js
+<Text>App name is: **{ asyncapi.info().title() }**</Text>
+```
+
+Newer:
+
+```js
+<Text>App name is: **{ asyncapi.info().title() }**</Text>
+<Text>Version is: **{params.version || asyncapi.info.version()}**</Text>
+```
+
+Now that you have added all the configuration options, you can start the generation process by using generator CLI. You can pass these parameters via the CLI. When using CLI, it's done by indicating --param name=value or -p name=value.
+As you can see, the above configuration will help the template user to override the existing version with a new version on the command line. For example, `-p version=2.0.0`.
+
+## Hooks
+
+[Hooks](hooks.md) enable templates to perform multiple tasks. You can add Hooks to your template as fractions of code. In the template, you must store it in the `hooks` directory under the template directory. You can also store it in other modules and external libraries, or even configure it inside the template. The generation process can perform multiple actions.
+
+**Templates** can perform multiple actions _before_ or _after_ the generation process with the help of **hooks**.
+
+For example, hooks can help you change the specification version with the new `version` that you can pass before the generation process even begins:
+
+```js
+module.exports = {
+  'generate:before': ({ asyncapi, templateParams = {} }) => {
+    const version = templateParams.version || asyncapi.info().version();
+    asyncapi._json.info.version = version;
+  }
+};
+```
+This can be an even better alternative way of overriding the `version` parameter that we discussed in the previous section. Not only a markdown document will be generated but also the AsyncAPI file that was passed to the generator will be returned with the overwritten version.
```

Comment:
```suggestion
This can be an even better alternative way of overriding the `version` parameter that we discussed in the previous section. Not only a markdown document will be generated but also the AsyncAPI document that was passed to the generator will be returned with the overwritten version.
```

yeah, make sure you say AsyncAPI document, not file